### ISSUE TITLE
A question about the alpha update

### ISSUE THREAD

**Issue by BangguWu** [MandyMo/DARTS#4]
I read the code of the train_search.py carefully, and I found that when calculating Equation 6 in the paper, you used L_val calculated by the un-updated model parameters, `L_val(w^', alpha)` instead of `w`.

In the code, I think w_t should be used to calculate valid_loss instead of  
```
valid_logits = self.model(valid_input)
```
 Please advise me if my understanding is correct? Thank you.

**Comment by MandyMo** [comment]
It seems that, the naive first-order gradient method still work pretty well, thus several successive works only exploit the naive SGD optimizer, e.g. FBnet, ProxylessNAS.
